docs/logout.rst

```diff
@@ -0,0 +1,162 @@
+local accounts. If you're using `Flask-Login`_ (or `Flask-Security`_,
+which is built on top of Flask-Login), you can import and call the
+:func:`flask_login.logout_user` function, like this:
+
+.. code-block:: python
+
+    from flask_login import logout_user
+    # other imports as necessary
+
+    @app.route("/logout")
+    def logout():
+        logout_user()
+        return redirect(somewhere)
+
+After you do this, your application will treat the user like any other
+anonymous user. However, logging out your user from their local account
+doesn't do anything about the provider account. As a result, if the
+user tries to log in again after you've logged them out this way,
+the conversation will look like this:
+
+1.  User: Hey CustomSite, I'm user ShinyStar99. Let me in.
+    Ask Google if you don't believe me.
+2.  CustomSite: Hey Google, who is this user?
+3.  Google: My user already gave me permission to tell you some
+    basic information. On *my* website, this user has the ID 987654.
+4.  CustomSite: Oh right, it's ShinyStar99 again. Go ahead.
+
+In most cases, this is what you want. However, sometimes you want to
+really reset things back to the start, as though the user had never
+granted consent to share information in the first place.
+
+Revoking Access with the Provider
+---------------------------------
+
+Undoing the user's permission to share information from the provider
+to your custom site is called "revoking access". Unfortunately,
+every provider has a different way of doing this, so you'll need
+to check the OAuth documentation provided by your OAuth provider.
+
+When you are granted access by a user, the provider will give your
+application a "token" that is used for making subsequent API requests.
+In order to revoke access for a user, you may need to include this
+token as an argument, so the provider knows which token to revoke.
+You can get this information by checking the ``token`` property of the
+Flask-Dance blueprint.
+
+We'll use Google as an example. First, check
+`Google's documentation for how to revoke access via OAuth2
+<https://developers.google.com/identity/protocols/OAuth2WebServer#tokenrevoke>`_.
+Notice that you do indeed need to provide the token in order to revoke it.
+
+Here's some sample code that works with Google:
+
+.. code-block:: python
+
+    from flask import Flask, redirect
+    from flask_dance.contrib.google import make_google_blueprint, google
+    from flask_login import logout_user
+
+    app = Flask(__name__)
+    blueprint = make_google_blueprint()
+    app.register_blueprint(blueprint, url_prefix="/login")
+
+    @app.route("/logout")
+    def logout():
+        token = blueprint.token["access_token"]
+        resp = google.post(
+            "https://accounts.google.com/o/oauth2/revoke",
+            params={"token": token},
+            headers={"Content-Type": "application/x-www-form-urlencoded"}
+        )
+        assert resp.ok, resp.text
```

Comment:
I'm a bit worried about this example. `assert` gets compiled out if you'd start python with `-O` which people might do in production.

I'm also not sure I'd want to break if revoking the access grant failed, in this case that would raise an `AssertionError` that's not handled resulting in a internal server error instead. I'm fine with keeping it as an example but I think it's worth documenting that in reality you should probably handle the case of `!resp.ok` in some gracefull manner instead.

Reply:
Good point. I added a warning about the `assert` statement. Do you think that's enough?